ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/roles.j2

```diff
@@ -0,0 +1,21 @@
+{% if roles is arista.avd.defined %}
+
+## Roles
+
+### Roles Summary
+{%     for role in roles | arista.avd.natural_sort %}
+#### Role {{ role.name }}
+
+| Sequence | Action | Mode | Command |
+| -------- | ------ | ---- | ------- |
+{%         for sequence in role.sequence_numbers | arista.avd.natural_sort %}
```

Comment:
please change `| arista.avd.natural_sort` to `| arista.avd.natural_sort('sequence')`. This will make the sort use the `sequence` field for sorting.